data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3SelectObjectWorker.java

```diff
@@ -0,0 +1,236 @@
+    private final BiConsumer<Event, S3ObjectReference> eventConsumer;
+    private final ObjectMapper mapper = new ObjectMapper();
+    private final JsonFactory jsonFactory = new JsonFactory();
+    public S3SelectObjectWorker(final S3ObjectRequest s3ObjectRequest) {
+        this.buffer = s3ObjectRequest.getBuffer();
+        this.numberOfRecordsToAccumulate = s3ObjectRequest.getNumberOfRecordsToAccumulate();
+        this.bufferTimeout = s3ObjectRequest.getBufferTimeout();
+        this.queryStatement = s3ObjectRequest.getQueryStatement();
+        this.serializationFormatOption = s3ObjectRequest.getSerializationFormatOption();
+        this.s3AsyncClient = s3ObjectRequest.getS3AsyncClient();
+        this.s3SelectResponseHandler = s3ObjectRequest.getS3SelectResponseHandler();
+        this.s3ObjectPluginMetrics = s3ObjectRequest.getS3ObjectPluginMetrics();
+        this.compressionType = s3ObjectRequest.getCompressionType();
+        this.fileHeaderInfo = s3ObjectRequest.getFileHeaderInfo();
+        this.eventConsumer = s3ObjectRequest.getEventConsumer();
+    }
+
+    /**
+     * Fetch the S3 object content using s3 select service and parsing and pushing to buffer.
+     * @param s3ObjectReference Contains bucket and s3 object details
+     *
+     * @throws IOException When Input/Output exception occur while processing the data.
+     */
+    public void parseS3Object(final S3ObjectReference s3ObjectReference) throws IOException {
+        try{
+            selectObjectContent(s3ObjectReference);
+        }catch(Exception e){
+            s3ObjectPluginMetrics.getS3ObjectsFailedCounter().increment();
+            throw new IOException(e);
+        }
+    }
+
+    private void selectObjectContent(S3ObjectReference s3ObjectReference) throws IOException {
+        final long totalBytesRead;
+        final InputStream inputStreamList;
+        final BufferAccumulator<Record<Event>> bufferAccumulator = BufferAccumulator.create(buffer, numberOfRecordsToAccumulate, bufferTimeout);
+        final InputSerialization inputSerialization = getInputSerializationFormat(serializationFormatOption
+                ,compressionType,fileHeaderInfo);
+        final SelectObjectContentRequest selectObjectContentRequest = getS3SelectObjRequest(s3ObjectReference, inputSerialization);
+        s3AsyncClient.selectObjectContent(selectObjectContentRequest, s3SelectResponseHandler).join();
+        final List<SelectObjectContentEventStream> receivedEvents = s3SelectResponseHandler.getReceivedEvents();
+        if (receivedEvents.isEmpty()) {
+            throw new IOException("No Records Found.");
+        }
+        inputStreamList = getInputStreamFromResponseHeader(s3SelectResponseHandler);
+                totalBytesRead = parseCompleteStreamFromResponseHeader(s3ObjectReference, bufferAccumulator,
+                        inputStreamList);
+
+        s3ObjectPluginMetrics.getS3ObjectSizeProcessedSummary().record(totalBytesRead);
```

Comment:
I think we should not populate this metric. This is not the actual object size. It is the size of the S3 Select response, which is different.